pkg/webhook/rollout/validating/rollout_create_update_handler.go

```diff
@@ -203,15 +203,23 @@ func validateRolloutSpecCanaryTraffic(traffic *appsv1alpha1.TrafficRouting, fldP
 	return errList
 }
 
-func validateRolloutSpecCanarySteps(steps []appsv1alpha1.CanaryStep, fldPath *field.Path, isTraffic bool) field.ErrorList {
+func validateRolloutSpecCanarySteps(steps []appsv1alpha1.CanaryStep, fldPath *field.Path, routings []*appsv1alpha1.TrafficRouting) field.ErrorList {
 	stepCount := len(steps)
 	if stepCount == 0 {
 		return field.ErrorList{field.Invalid(fldPath, steps, "The number of Canary.Steps cannot be empty")}
 	}
 
+	var isTraffic bool
+	var ingressClassType string
+	if len(routings) > 0 {
+		isTraffic = true
+		if routings[0].Ingress != nil {
+			ingressClassType = routings[0].Ingress.ClassType
+		}
+	}
 	for i := range steps {
 		s := &steps[i]
-		if isTraffic && s.Weight == nil && len(s.Matches) == 0 {
+		if isTraffic && s.Weight == nil && len(s.Matches) == 0 && ingressClassType != "aliyun-alb" {
```

Comment:
what's different about aliyun-alb, plz comment in the code. in general one should not hard code such provider type in the code

Reply:
sorry, this code has removed